src/components/MoviesSection/PaginationComponents/PaginationComponent.tsx

```diff
@@ -1,37 +1,65 @@
-import React, { FC, useCallback, useState } from 'react';
+import React, { FC, useCallback } from 'react';
 import { Pagination, PaginationItem, PaginationLink } from 'reactstrap';
+import { useDispatch, useSelector } from 'react-redux';
 import { PaginationItemWithLink } from './PaginationItemWithLink';
 import './Pagination.scss';
+import { moviesDataSelector } from '../../../redux/selector/selectors';
+import store from '../../../store';
+import { setCurrentPageCreator } from '../../../redux/actions/actionCreators/actionCreators';
 
 export const PaginationComponent: FC = () => {
-  const [currentPage, setPage] = useState(1);
-  const [totalPages] = useState(8);
-  // just mock data that will be updated with Redux in the next tasks
+
+  let { totalPages, totalAmount } = useSelector(moviesDataSelector);
+  const { currentPage } = useSelector(moviesDataSelector);
+  const dispatch = useDispatch();
+
+  store.subscribe(() => {
+    totalPages = store.getState().movieReducer.totalPages;
+    totalAmount = store.getState().movieReducer.totalAmount;
+  });
 
   const changePage = useCallback((page: number) => {
     if (page > 0 && page <= totalPages) {
-      setPage(page);
+      dispatch(setCurrentPageCreator(page));
     }
   }, []);
 
-  const pages = [];
-  for (let i = 1; i <= totalPages; i++) {
-    pages.push(<PaginationItemWithLink
-      key={i}
-      itemInfo={{
-        pageKey: i,
-        active: currentPage === i,
-      }}
-      callBack={changePage}
-    />);
-  }
+  const range = (currentPage: number, totalPages: number) => {
+    if (totalPages === 0) {
+      return [];
+    }
+    let start = 0;
+    let end = 0;
+    if (currentPage === 1) {
+      start = 1;
+      end = Math.min(totalPages + 1, 9);
+    } else if (currentPage < 8) {
+      start = 1;
+      end = Math.min(9, totalPages + 1);
+    } else {
+      start = currentPage;
+      end = Math.min(start + 8, totalPages + 1);
+    }
+    return Array.from(Array.from(Array(Math.ceil(end - start)).keys()), x => start + x);
```

Comment:
Looks overcomplicate 
Could we simplify it?

Reply:
Simplified it in commit e9860952613603eb2cf3095f28c180d207b9668c